packages/react/src/bags/hooks/useBag.ts

```diff
@@ -159,7 +160,16 @@ const useBag = (options: UseBagOptions = {}) => {
   );
 
   useEffect(() => {
-    if (!isLoading && !isFetched && enableAutoFetch && userBagId) {
+    const updatedState = store.getState();
+    const updatedIsLoading = isBagLoading(updatedState as StoreState);
```

Comment:
Instead of casting here, cast the `updatedState` variable to `StoreState` type like this:

```suggestion
    const updatedState = store.getState() as StoreState;
    const updatedIsLoading = isBagLoading(updatedState);
    const updatedIsFetched = isBagFetched(updatedState);
```

Reply:
The same applies to the other hooks